pkg/pods/manifest.go

```diff
@@ -320,3 +324,22 @@ func (m manifest) SignatureData() (plaintext, signature []byte) {
 	}
 	return m.plaintext, m.signature
 }
+
+// ValidManifest checks the internal consistency of a manifest. Returns an error if the
+// data is inconsistent or "nil" otherwise.
+func ValidManifest(m Manifest) error {
+	if m.ID() == "" {
+		return fmt.Errorf("manifest must contain an 'id'")
+	}
+	for key, stanza := range m.GetLaunchableStanzas() {
+		switch {
+		case stanza.LaunchableType == "":
+			return fmt.Errorf("'%s': launchable must contain a 'launchable_type'", key)
+		case stanza.LaunchableId == "":
```

Comment:
Currently, the key is ignored and `LaunchableId` is used everywhere.  Once that changes, we can alter the validation routine here.